src/Microsoft.ML.Data/Evaluators/RegressionEvaluator.cs

```diff
@@ -102,7 +102,16 @@ public override double RSquared
                 {
                     get
                     {
-                        return SumWeights > 0 ? 1 - TotalL2Loss / (TotalLabelSquaredW - TotalLabelW * TotalLabelW / SumWeights) : 0;
+                        // RSquared value cannot be well-defined with less than two samples.
+                        // Return NaN instead of -Infinity.
+                        if (SumWeights > 0)
+                        {
+                            if ((TotalLabelSquaredW - TotalLabelW * TotalLabelW / SumWeights) == 0)
+                                return double.NaN;
+                            return 1 - TotalL2Loss / (TotalLabelSquaredW - TotalLabelW * TotalLabelW / SumWeights);
+                        }
+                        else
```

Comment:
nit: Can we get rid of the ` else` and just return 0?

Reply:
Fixed!